pkg/schemapatcher/gen.go

```diff
@@ -142,6 +147,12 @@ func (g Generator) Generate(ctx *genall.GenerationContext) (result error) {
 				fullSchema = *fullSchema.DeepCopy()
 				crdgen.TruncateDescription(&fullSchema, *g.MaxDescLen)
 			}
+
+			// Fix top level ObjectMeta regardless of the settings.
+			if _, ok := fullSchema.Properties["metadata"]; ok {
```

Comment:
this doesn't seem quite right, because top-level metadata is actually allowed to contain a couple validations, and this is one of the things schemapatch enables.

Reply:
Interesting, I basically copied the approach over from the `crd` generator, where this exact change is done unconditionally as well :thinking:. 

See

https://github.com/kubernetes-sigs/controller-tools/blob/97ee66cc83b03bd632cf8da35496812ca091ae4e/pkg/crd/gen.go#L137-L138

and

https://github.com/kubernetes-sigs/controller-tools/blob/97ee66cc83b03bd632cf8da35496812ca091ae4e/pkg/crd/gen.go#L281-L290

Could you elaborate on why this should be different in `schemapatch`?

Reply:
oh hmm this might've been me reviewing too fast.  My thought was that maybe people had put in place custom validation, but schemapatch is gonna overwrite that anyway.  nevermind :-)